### ISSUE TITLE
Sending activity level and heart beat rate to nightscout

### ISSUE THREAD

**Issue by pooyaww** [NightscoutFoundation/xDrip#1857]
In xDrip it is possible to collect these data, it would be nice if we could have them on Nightsout also.
Activity level, in particular for children, is an important factor for low predictions.
Besides, heart beat level is also important to decide on the current state of activities.

**Comment by Navid200** [comment]
@tolot27 I see this closely related to the following.
https://github.com/NightscoutFoundation/xDrip/issues/1476
Of course, this one wants to upload to Nightscout and other wants to upload to tidepool.  Someone may want to open another one for uploading the same to another server.  
But, what is common between all of these is what is requested from xDrip to upload.

If the person who has to approve uploading such data doesn't feel that the data should be uploaded (I'm not saying that's the case), he/she is not going to approve any of them.  Is there really a benefit in having more requests?
In my opinion, we should combine related requests.  If some day a developer creates a PR for only uploading to Nightscout but not to tidepool, the combined issue will remain open because it hasn't been addressed fully.

So, if you agree, I would ask @pooyaww to add their request to the linked existing issue, and close this one.

**Comment by Navid200** [comment]
@pooyaww Thanks

**Comment by tolot27** [comment]
> But, what is common between all of these is what is requested from xDrip to upload.
> In my opinion, we should combine related requests. If some day a developer creates a PR for only uploading to Nightscout but not to tidepool, the combined issue will remain open because it hasn't been addressed fully.

@Navid I can follow you intention and already thought about refactoring the different sync features into a centralized sync framework. That would allow us to implement the features for all sync targets at the same time. But as long as we have target-specific issues, we should keep it separate. Otherwise, comments will be mixed and it is hard to follow.

> So, if you agree, I would ask @pooyaww to add their request to the linked existing issue, and close this one.

Please keep the issues separated by the sync target/source.

**Comment by tolot27** [comment]
@pooyaww It looks like the data can be synced if engineering mode is enabled. Did you try that? What is the source of the heart beat? A pebble watch or did you enable the "Smartwatch Sensors" in xDrip?

The feature should be easy to implement.

**Comment by pooyaww** [comment]
@tolot27 How can I enable Engineering mode? In my phone it is grayed out.
The source of the heart beat is a Fitbit Versa 3 , and I have enabled Smartwatch Sensor in xDrip.

**Comment by Navid200** [comment]
@pooyaww https://github.com/Navid200/xDrip/wiki/Engineering-mode

**Comment by pooyaww** [comment]
@tolot27 I enabled engineering mode, but nothing happened about activity level.

**Comment by tolot27** [comment]
Please can you post a screenshot which shows the data collected in xDrip?

Please also enable extra logging tag for "NightscoutUploader:v" and report, if you see anything like "Processing (heartrate|steps|motion tracking) for RESTAPI" over time in the logs.
